From 0361bd4da12e79fb9fb770ea1c7c5dd301078e88 Mon Sep 17 00:00:00 2001
From: Huynh Thanh Hung <hung.huynh.wz@renesas.com>
Date: Fri, 9 Oct 2020 10:07:42 +0700
Subject: [PATCH] core: arm: plat-rzg: platform-config: Re-configure memory for
 RZ/G2H

Previous we miss define for ECC partial, so that when use ECC partial,
it will use the memory for non-ECC, that have wrong DRAM2 start address.

Now we re-configure that cover all ECC setting.

Signed-off-by: Huynh Thanh Hung <hung.huynh.wz@renesas.com>
---
 core/arch/arm/plat-rzg/platform_config.h | 21 ++++++++++++++++-----
 1 file changed, 16 insertions(+), 5 deletions(-)

diff --git a/core/arch/arm/plat-rzg/platform_config.h b/core/arch/arm/plat-rzg/platform_config.h
index d47f6626..8f967617 100644
--- a/core/arch/arm/plat-rzg/platform_config.h
+++ b/core/arch/arm/plat-rzg/platform_config.h
@@ -82,21 +82,32 @@
 #endif
 
 #if defined(PLATFORM_FLAVOR_hihope_rzg2h)
-#if (RZG_DRAM_ECC == 1 && RZG_ECC_FULL == 1)	//Apply ECC Full Dual setting
+#if (RZG_DRAM_ECC == 1)
+#if (RZG_ECC_FULL == 1) //Apply ECC Full Dual setting
+/* Define DDR0 memory map only*/
 #define NSEC_DDR_0_BASE     0x47E00000U
 #define NSEC_DDR_0_SIZE     0x78200000
-#elif (RZG_DRAM_ECC == 1 && RZG_ECC_FULL == 2)	//Apply ECC Full Single setting
+#elif (RZG_ECC_FULL == 2) //Apply ECC Full Single setting
+/* Define half DDR0/DDR1 memory map*/
 #define NSEC_DDR_0_BASE     0x47E00000U
 #define NSEC_DDR_0_SIZE     0x4C200000
 #define NSEC_DDR_1_BASE     0x600000000U
 #define NSEC_DDR_1_SIZE     0x40000000
-#else
+#else //(RZG_ECC_FULL) //Apply ECC Partial setting
+/* Define full of DDR0/DDR1 memory map*/
 #define NSEC_DDR_0_BASE     0x47E00000U
+#define NSEC_DDR_1_BASE     0x600000000U
 #define NSEC_DDR_0_SIZE     0x78200000
+#define NSEC_DDR_1_SIZE     0x80000000
+#endif //(RZG_ECC_FULL)
+#else //(RZG_DRAM_ECC)
+/* Define original memory*/
+#define NSEC_DDR_0_BASE     0x47E00000U
 #define NSEC_DDR_1_BASE     0x500000000U
+#define NSEC_DDR_0_SIZE     0x78200000
 #define NSEC_DDR_1_SIZE     0x80000000
-#endif //(RZG_DRAM_ECC == 1 && RZG_ECC_FULL == 1)
-#endif
+#endif //(RZG_DRAM_ECC)
+#endif //(PLATFORM_FLAVOR_hihope_rzg2h)
 
 /* Full GlobalPlatform test suite requires TEE_SHMEM_SIZE to be at least 2MB */
 #define TEE_SHMEM_START     (TZDRAM_BASE + TZDRAM_SIZE)
-- 
2.17.1

